doc/contributing.md

```diff
@@ -0,0 +1,247 @@
+relevent to your contribution without harming the progress of others. This 
+duplicate is known as your repository and every team member has a fork. This is 
+different from the main repository which belongs to the team rather than an 
+individual. Don't confuse the main repository with the master branch. The
+shared, main repository can have incomplete features being worked on by 
+everyone. The master in your fork and the master in the main repository should
+always remain clean.
+
+This also works well when you want a feature that only some people want. You 
+don't have to move every contribution from your fork back into the main 
+repository. When you have contributed something and want it placed in master, 
+whoever is reviewing your code will look at your fork of the code and compare 
+it to the master branch.
+
+### Branches
+Branches are like pieces of scrap paper. You can use them to organize your work 
+and solutions to the test questions. You should not have work regarding 
+different problems mixed across several pieces of paper; you may get your 
+progress confused. You should use one (or several) peices of paper for each
+problem you are trying to solve, but should never use one sheet for multiple
+questions.
+
+For RoboCup you should create a new branch for every new item you'd like to work
+on and for every bug or issue you have to fix. This ensures your master branch
+stays clean. You should never solve more than one issue at a time and you should
+never have changes or additions for multiple things in the same branch. You can 
+look at a typical branching layout ![branchingModel](http://justinhileman.info/article/changing-history/git-flow.png).
+
+### Remotes
+If Git is like a test, then remotes would be cheating. A remote allows you to
+view the solution(s) of another classmate, and pull those additions into your 
+repostory as if they were on the classroom whiteboard.
+
+In software this can be particularly useful if a team member is working on some
+new code that may not be perfect yet, but isn't ready to be folded into master. 
+This may happen when you cannot continue work without getting the somewhat 
+related progress from someone else. You should understand that when pulling from 
+someone else, they take no responsibility for any problems their updates may
+cause for you. This is a decently advanced concept for those new to distributed 
+environments, and won't be used too often. We encourage you learn more about 
+this independently if interested. 
+
+### Overall Workflow (important)
+If you read the previous sections, you may be a little overwhelmed. This section
+will describe how these elements interact to form a coherent workflow that will 
+allow you to make contributions more easily. You can view an overall diagram of 
+how data moves between team members and GitHub ![githubDataFlow](http://www.dalescott.net/wp-content/uploads/2012/09/centralized-github-4.png).
```

Comment:
This one too.  Put the period first and try putting the image on its own line below the paragraph.